### DISCUSSION TITLE
Limit the number of POIs

### DISCUSSION THREAD

**Discussion by Hoerli1337** [osmandapp/OsmAnd#12552] [Q&A] [answered]
Hello!
Either I'm blind or the function does not exist :)
Is there a setting where I can limit the number of POIs currently displayed on the screen?
Or the POIs are only displayed after a certain zoom level?
When I am a tourist and looking for places, I turn on many POIs. But then it's almost impossible to see the place names or streets, because it feels like 1 million POIs are displayed.
I can turn them off manually, but sometimes there are important ones that I would like to see.
As a tourist you do not know and look for everything helpful :)
If that does not exist, it would be a good extension.

Google, Here or Sygic already do that quite well.
How does it work with OsmAnd?
(Have the F-Droid version)
![Screenshot_20210821-113422_OsmAnd~](https://user-images.githubusercontent.com/13108516/130317673-83adcad2-f3f5-408a-8487-c7304479df4b.png)
![Screenshot_20210821-113500_OsmAnd~](https://user-images.githubusercontent.com/13108516/130317685-43901e92-509d-467c-b618-088fb55a3774.png)

**Comment by pebogufi** [comment]
The orange circles are the result of a search, which YOU selected. 
Why don't you change the search condition, if you like to see other / less result objects ?

**Reply by Hoerli1337** [reply]
The orange circles are not a search, they are just normal POIs that I show. I have not yet searched for anything.
But I also want to be able to explore places and not end up in the display chaos.
To plan a route, it can be quite advantageous to see a bank and next a restaurant to eat where the hiking trail starts right next to it.
If everything is displayed at once, it is actually just stupid or do you have a view on photo 1? I don't ;)
Other map apps can display this more nicely without really having an information problem.

I received the same feedback from other people who I had told that OsmAnd is actually a great app.
They were also surprised by the current maps, but the amount of information was then suddenly too much.
Therefore, I have thrown the question here times in space. Apparently it is not only me so.

**Comment by tommycrock** [comment, marked as answer]
Did you know you can turn the labels off (i.e. just symbols) which would declutter a little.

> I can turn them off manually, but sometimes there are important ones that I would like to see.
> As a tourist you do not know and look for everything helpful :)

Do you have any criteria you suggest osmand should filter POIs by? 

I might be frustrated to think I've selected to be shown certain POIs but then discover they were hidden.

**Reply by Hoerli1337** [reply]
No, just maybe limit the amount first. No matter what it is.
Maybe display 10 POIs at once and the further I zoom into the map, the more will be displayed.
I want to be able to read at least the names of the places.
If I hide some POIs (e.g. all benches or bus stops) more restaurants or hospitals can appear.
The confusion in photo 1 is just stupid. You can't see anything of the map anymore.

**Reply by tommycrock** [reply]
I could see the advantage in clustering POIs, to display one symbol when one type got so dense it was overlapping. Then click on the symbol and it lists nearby ones or jumps up and shows fine lines to the individual POIs. But if I've asked it to show me restaurants and it's hidden them because I'm also showing benches and eight other things, that would be annoying. You'd find yourself constantly zooming in and out to see if you'd missed something and never be sure what remained hidden.

**Reply by Hoerli1337** [reply]
That is true.
But what would have to be adjusted urgently would be at least the level of the text to the symbol.
Names of cities should be on the top level and should not be covered by the icons.
Let's hope that one of the UI designers reads the problem here one day and has the great idea for a sensible implementation.
Then using the app while exploring the environment will be a bit more fun.
=> I find it anyway great that you can save everything offline :)

**Reply by tommycrock** [reply]
I'm wondering if you have POIs and labels as displayed by default hidden? They don't have the orange circles but look like this:
![Screenshot_20210821-231036_OsmAnd+](https://user-images.githubusercontent.com/5749884/130336077-9bdb0e42-a26b-4380-ac9e-d22f2a7ea81a.jpg)
They automatically (dis)appear as you zoom. Try configure map> Map rendering: Hide...

**Reply by Hoerli1337** [reply]
Ohhh there the thing has me completely confused.
POI overlay = The icons in orange completely prioritized and always displayed.
POI overlay designation = The normal icons as you need them.
The latter was disabled by default on 3 devices for me.
Habs now activated and now you can use it reasonably.
Now it also looks like yours.

**Reply by pebogufi** [reply]
@Hoerli1337 The "POI Overlay" and (result of) Search Categories is identical !
It is just a different user access. Try out what Search can afford, have a look on user defined search, which is also visible in POI Overlay. 
User defined search is a VERY powerful feature of osmand !

**Reply by Lee-Carre** [reply]
@Hoerli1337

There are multiple ways to enable PoI highlighting (overlay). The intended use is to emphasise *specific* types of extra interest (which may not be shown by default, especially small features). Not *all* of them at once 😮.

I noticed, in one of your display-captures, that you seem to have Mapillary and the overlay opacity slider enabled, too. Enabling all of OsmAnd's (*many*) features at the same time is unwise and will lead to the kind of chaos you've already seen.

Different tools, at different times, for different tasks. This isn't your grandfather's map 😉.

Start with the defaults, and only tweak where necessary. The default behaviour, for PoI, is exactly as you describe your preferred behaviour. It seems like you somehow enabled PoI highlighting for every type of PoI (which, effectively, disables zoom-based filtering, and colour-differentiation).

You'd be much better off searching / highlighting specific things as & when you need.
Even with only a few categories enabled, the highlighting can become overwhelming (I speak from experience 🙂).

A really simple & clear use-case for PoI highlighting would be when using the car profile and driving a (battery-)EV, having amenity=charging_station highlighted. Either normally (any time the car profile is active) or only during navigation (with the car profile, of course) via the show-along-the-route feature (which limits the (user-configurable) radius of those shown, rather than showing **all** of them, everywhere, indiscriminately). Besides being of particular importance to (battery-)EV drivers, charging stations are also small features, and wouldn't necessarily be rendered by default, especially if there are many other PoIs in the area.
With amenity=charging_station highlighted, they're *always* shown, regardless of other PoIs, and always-on-top.

The details change for other situations, but hopefully that's enough to give you the gist of the concept.

You mentioned being a tourist; I think there even a category (for easy selection) of tourist-interest PoIs.

You might also want to consider enabling (& downloading) wikivoyage guides.

Seems like you're in a well-mapped place, so OsmAnd should work even better for you, when used right.

**Reply by SomeoneSerge** [reply]
> Do you have any criteria you suggest osmand should filter POIs by?

A trivial baseline to start from: select up to `N` POIs from the current scale arbitrarily; as long as POIs are in the view, keep them "selected"; when a POI leaves the view, "deselect"; when an unselected POI enters the view and we're displaying less than `N` POIs, select it; allow the user to control `N`

**Comment by vshcherb** [comment]
Screenshot looks like a problem of 3.9 there were fixes with 4.0

**Comment by ThomasMeister123** [comment]
Tip for the developers: Look how LocusMaps handles POIs in different zoom-leves (grouoing!!!)
Waiting for this feature in OSMAND for years!
